resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/submit/Client.scala

```diff
@@ -266,30 +284,36 @@ private[spark] class Client(
   }
 }
 
-private[spark] object Client {
+private[spark] object Client{
   def main(args: Array[String]): Unit = {
     val sparkConf = new SparkConf(true)
     val mainAppResource = args(0)
     val mainClass = args(1)
     val appArgs = args.drop(2)
     run(sparkConf, mainAppResource, mainClass, appArgs)
   }
-
   def run(
       sparkConf: SparkConf,
       mainAppResource: String,
       mainClass: String,
       appArgs: Array[String]): Unit = {
-    val sparkJars = sparkConf.getOption("spark.jars")
+    val isPython = mainAppResource.endsWith(".py")
+    val sparkJars = if (isPython) Array.empty[String] else {
+      sparkConf.getOption("spark.jars")
       .map(_.split(","))
       .getOrElse(Array.empty[String]) ++
       Option(mainAppResource)
         .filterNot(_ == SparkLauncher.NO_RESOURCE)
-        .toSeq
+        .toSeq }
     val launchTime = System.currentTimeMillis
     val sparkFiles = sparkConf.getOption("spark.files")
       .map(_.split(","))
       .getOrElse(Array.empty[String])
+    val pySparkFiles: Array[String] = if (isPython) {
+      appArgs(0) match {
+        case null => Array(mainAppResource)
```

Comment:
keying in on a `null` return is a scala no-no.  Recommend `if (appArgs.isEmpty) ...` or some other test